python/semantic_kernel/orchestration/sk_function.py

```diff
@@ -166,9 +183,24 @@ async def _local_func(client, request_settings, context: "SKContext", **kwargs):
                     if function_call is not None:
                         context.objects["function_call"] = function_call
                 else:
-                    completion = await client.complete_chat_async(
-                        messages, request_settings
-                    )
+                    if request_settings.data_source_settings is not None and hasattr(
+                        client, "complete_chat_with_data_async"
```

Comment:
What happens with OpenAI Client classes with this?

Reply:
Switched the order so the check for with_data method is first, I think this should be safe for OpenAI clients but I don't have credentials to test one manually